specification/hardwaresecuritymodules/HardwareSecurityModules.Management/models.tsp

```diff
@@ -0,0 +1,851 @@
+  Application: "Application",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  ManagedIdentity: "ManagedIdentity",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  Key: "Key",
+}
+
+/**
+ * Sku name of the Cloud HSM Cluster
+ */
+#suppress "@azure-tools/typespec-azure-core/no-enum" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+enum CloudHsmClusterSkuName {
+  Standard_B1,
+  `Standard B10`,
+}
+
+/**
+ * Properties of a Cloud HSM Cluster.
+ */
+model CloudHsmClusterProperties {
+  /**
+   * State of security domain activation
+   */
+  @visibility(Lifecycle.Read)
+  activationState?: ActivationState;
+
+  /**
+   * The Cloud HSM Cluster's auto-generated Domain Name Label Scope
+   */
+  autoGeneratedDomainNameLabelScope?: AutoGeneratedDomainNameLabelScope;
+
+  /**
+   * An array of Cloud HSM Cluster's HSMs
+   */
+  @visibility(Lifecycle.Read)
+  @OpenAPI.extension("x-ms-identifiers", #["fqdn"])
+  hsms?: CloudHsmProperties[];
+
+  /**
+   * List of private endpoint connection resources
+   */
+  @visibility(Lifecycle.Read)
+  @OpenAPI.extension("x-ms-identifiers", #["name"])
+  privateEndpointConnections?: PrivateEndpointConnection[];
+
+  /**
+   * The Cloud HSM Cluster's provisioningState
+   */
+  @visibility(Lifecycle.Read)
+  provisioningState?: ProvisioningState;
+
+  /**
+   * The Cloud HSM Cluster public network access
+   */
+  publicNetworkAccess?: PublicNetworkAccess;
+
+  /**
+   * Cloud HSM Cluster status message
+   */
+  @visibility(Lifecycle.Read)
+  statusMessage?: string;
+}
+
+/**
+ * The Cloud HSM Properties
+ */
+model CloudHsmProperties {
+  /**
+   * FQDN of the Cloud HSM
+   */
+  fqdn?: string;
+
+  /**
+   * The Cloud HSM State. Values are: Deploying, ConfiguringSlb, Starting, Starting, Failed, Failed, Deleting, DeletingSlbEntry, InitialProvisioning, Updating
+   */
+  state?: string;
+
+  /**
+   * The Cloud HSM State message
+   */
+  stateMessage?: string;
+}
+
+/**
+ * Properties of the private endpoint connection.
+ */
+model PrivateEndpointConnectionProperties {
+  /**
+   * The private endpoint resource.
+   */
+  privateEndpoint?: CommonTypes.PrivateEndpoint;
+
+  /**
+   * A collection of information about the state of the connection between service consumer and provider.
+   */
+  privateLinkServiceConnectionState: CommonTypes.PrivateLinkServiceConnectionState;
+
+  /**
+   * The provisioning state of the private endpoint connection resource.
+   */
+  @visibility(Lifecycle.Read)
+  provisioningState?: PrivateEndpointConnectionProvisioningState;
+
+  /**
+   * The group ids for the private endpoint resource.
+   */
+  @visibility(Lifecycle.Read)
+  groupIds?: string[];
+}
```

Comment:
Private endpoint connections being a standard contract across many RPs, should be modeled with a common types contract - can we do it?